shell/server/pre-meteor.js

```diff
@@ -90,7 +96,15 @@ Meteor.startup(function () {
       publicIdPromise = Promise.resolve(id);
     } else {
       // Not a wildcard host. Perhaps it is a custom host.
-      publicIdPromise = lookupPublicIdFromDns(hostname);
+
+      // Skip DNS TXT record resolutions for the major pseudo-TLDs, and avoid
+      // returning any result which might indicate that there's actually a
+      // server listening here.
+      if (hasPseudoTLD(hostname)) {
+        return;
+      } else {
+        publicIdPromise = lookupPublicIdFromDns(hostname);
```

Comment:
Yes, I originally put the changes in `lookupPublicIdFromDns()` and returned a new Error there. However, I changed it because I didn't want the error handling in `Meteor.start()` to do something like return HTTP 500 with a message like *"Hey, Sandstorm server here, and we're responding to tell you that we're pretending not to know what `blahblahblahblahblah.onion` is!"*